### ISSUE TITLE
Bug: snuffleupagus requires quotes in ifs.

### ISSUE THREAD

**Issue by majkeldzordan** [jvoisin/snuffleupagus#432]
The latest version (previously there was no problem) has a strange bug. I accept the thought that this is kind of a patch, but:
1. should not be enabled by default
2. I have not found any info on how to turn it off.

Well. When I download the contents of the file (here: the `test.txt` file has only one character - a number, e.g. 0), the code:

```php
$variable = file_get_contents ("test.txt");
if ($variable == "0") // works fine
if ($variable == 0) // doesn't work
```

I am not sure if for the above mentioned reason there is also a problem with older versions of WordPress (I do not penetrate its code so thoroughly), eg 4.9.x. After logging into wp-admin, there is a "update database" loop - it cannot be done. Disabling snuffleupagus or downgrade also solves this problem.

**Comment by jvoisin** [comment]
Do you have `sp.sloppy_comparison.enable();` in your configuration file?

**Comment by majkeldzordan** [comment]
Configuration file is empty. Adding this parameter will fix the problem (I'm currently working on an older version to keep everything working)?

**Comment by jvoisin** [comment]
Can you try anyway?

**Comment by majkeldzordan** [comment]
I tested it. I updated snuffleupagus, added to the configuration file:
`sp.sloppy_comparison.enable ();`
and it still doesn't work.

**Comment by jvoisin** [comment]
I can reproduce the issue locally, there is indeed a bug in Snuffleupagus. I'm working on a fix.

**Comment by jvoisin** [comment]
Are you using PHP8?

**Comment by majkeldzordan** [comment]
I tested it on 7.4.

**Comment by jvoisin** [comment]
Should be fixed via e6219a095de78ce03be4a36360020a6f4fe94105
